doc/website/release-notes/iceoryx-unreleased.md

```diff
@@ -261,3 +262,50 @@
         // perform error handling
     }
     ```
+
+14. Remove `forEach` from helplets
+    ```cpp
+    // before
+    iox::cxx::forEach(container, [&] (element) { /* do stuff with element */ });
+
+    // after
+    for (const auto& element: container) { /* do stuff with element */ }
+    ```
+
+15. Remove `enumTypeAsUnderlyingType`
+    ```cpp
+    constexpr const char* SOME_ENUM_STRINGS[] = {"FOO", "BAR"};
+
+    // before
+    std::cout << SOME_ENUM_STRINGS[iox::cxx::enumTypeAsUnderlyingType(someEnum)] << std::endl;
+
+    // after
+    std::cout << SOME_ENUM_STRINGS[static_cast<uint64_t>(someEnum)] << std::endl;
+    ```
+
+16. Remove `convertEnumToString`
+    ```cpp
+    constexpr const char* SOME_ENUM_STRINGS[] = {"FOO", "BAR"};
+
+    // before
+    std::cout << iox::cxx::convertEnumToString(SOME_ENUM_STRINGS, someEnum] << std::endl;
+
+    // after
+    std::cout << SOME_ENUM_STRINGS[static_cast<uint64_t>(someEnum)] << std::endl;
+    ```
+
+17. Replace `strlen2` with more generic `arrayCapacity`
+    ```cpp
+    constexpr const char LITERAL1[] {"FOO"};
+    constexpr const char LITERAL2[20] {"BAR"};
+    constexpr const uint32_t ARRAY[42] {};
+
+    // before
+    std::cout <<  << iox::cxx::strlen2(LITERAL1) << std::endl; // prints 3
+    std::cout <<  << iox::cxx::strlen2(LITERAL2) << std::endl; // prints 20
+
+    // after
+    std::cout << arrayCapacity(LITERAL1) << std::endl; // prints 4
```

Comment:
Ähm.... [click here](https://pics.me.me/there-are-two-hard-problems-in-computer-science-0-naming-26556167.png)? :grin:

Reply:
Hahaha, I like that but in this case the 4 is correct. `"FOO"` is actually `['F', 'O', 'O', '\0']`. Therefore this was renamed to `arrayCapacity`. It is more generic and gives the actual array capacity

Reply:
@elBoberido And before that it was the case, that's why 3 was the result?

Reply:
@elBoberido before it was only meant for string literals and the `\0` was subtracted but that was error prone since one can also have `constexpr const char FOO[20] {"FOO"}` which would not return what one would expect. From my point of view it's less ambiguous to talk about array sizes. This has the additional benefit that it is more generic

Reply:
@mossmaurice but you were right with the off by one. `std::cout <<  << iox::cxx::strlen2(LITERAL2) << std::endl; // prints 20` should be 19